backend/ingestion/extract_scheduled_gtfs.py

```diff
@@ -0,0 +1,304 @@
+    # Perhaps pass NOW in from outside the function, so it's the same for all entries
+    # from a particular GTFS feed as it gets processed
+
+    # TODO: miscellaneous cleanup (one thing: standardize CTA route names to
+    # standard English words for colors)
+    # TODO: Consider adding a # sign to color column (or see if Postgres has a
+    # color data type)
+    return feed_component
+
+
+def combine_different_feeds(
+    feed_url_list: list[str], output_folder: None | str = None
+) -> dict[pd.DataFrame | gpd.GeoDataFrame]:
+    """
+    Run the entire GTFS ingestion pipeline for multiple endpoint URLs.
+    TODO: consider allowing this to take an arbitrary list of *args rather than
+    a list.
+
+    Optional parameter output_folder allows for saving results to file storage.
+    """
+    # This doesn't seem to require initailizing empty dataframes with the proper
+    # columns to concat to -- it'll take those from the first dataframe processed
+    all_routes = None
+    all_trips = None
+    all_stops = None
+    all_stop_times = None
+    all_shapes = None
+    all_transfers = None
+    all_shape_geometries = None
+
+    for this_url in feed_url_list:
+        this_city, this_agency, this_feed = ingest_gtfs_feed(this_url)
+        print(f"{this_city} {this_agency} feed ingested")
+        these_gtfs_dataframes = {
+            name: add_extra_columns(this_city, this_agency, df)
+            for name, df in get_gtfs_component_dfs(
+                this_city, this_agency, this_feed
+            ).items()
+        }
+
+        for name, df in these_gtfs_dataframes.items():
+            # TODO: this in a  more programmatic way
+            if name == "routes":
+                all_routes = pd.concat([all_routes, df])
+            elif name == "trips":
+                all_trips = pd.concat([all_trips, df])
+            elif name == "stops":
+                all_stops = pd.concat([all_stops, df])
+            elif name == "stop_times":
+                all_stop_times = pd.concat([all_stop_times, df])
+            elif name == "shapes":
+                all_shapes = pd.concat([all_shapes, df])
+            elif name == "transfers":
+                all_transfers = pd.concat([all_transfers, df])
+            elif name == "shape_geometries":
+                all_shape_geometries = pd.concat([all_shape_geometries, df])
+        print(f"{this_city} {this_agency} dataframes concatenated to whole")
+    # end for
+
+    all_dfs = {
+        "routes": all_routes,
+        "trips": all_trips,
+        "stops": all_stops,
+        "stop_times": all_stop_times,
+        "shapes": all_shapes,
+        "transfers": all_transfers,
+        "shape_geometries": all_shape_geometries,
+    }
+
+    print(all_dfs)
+
+    if output_folder is not None:
+        for name, df in all_dfs.items():
+            pd.to_csv(df, f"{output_folder}/{name}.csv")
+
+    return all_dfs
+
+
+# TODO: function to convert lat/long and shapes.txt data into geometry objects.
```

Comment:
👍 can help w using the [shapely](https://pypi.org/project/shapely/) tools if you end up going this route